libbeat/publisher/queue/diskqueue/consumer.go

```diff
@@ -0,0 +1,52 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package diskqueue
+
+import (
+	"github.com/elastic/beats/v7/libbeat/publisher"
+	"github.com/elastic/beats/v7/libbeat/publisher/queue"
+)
+
+type diskQueueConsumer struct {
```

Comment:
Well, it's an internal type, so it won't have the external context of a forced `diskqueue.Consumer`, we'll just see the bare type name inside this package. Having read a lot of this code lately, I prefer the explicit form to bare names like `consumer` and `delegate` and such, because when someone comes back to this code they need extra context / knowledge for the shorter name to make sense (of which package they're in, how it fits into the larger design hierarchy, etc). `diskQueueConsumer` is slightly longer to type, but anyone who sees it will get at least some idea of what it's pointing at.